### ISSUE TITLE
Ref

### ISSUE THREAD

**Issue by at15** [at15/snowbot#1]
- tf chatbot, stanford cs20si
  - https://web.stanford.edu/class/cs20si/lectures/slides_13.pdf 
  - http://web.stanford.edu/class/cs20si/assignments/a3.pdf
  - https://github.com/chiphuyen/stanford-tensorflow-tutorials/tree/master/assignments/chatbot 
- another tf chatbot  Seems to be encoder decoder? ...
  - https://github.com/llSourcell/tensorflow_chatbot
  - https://github.com/suriyadeepan/practical_seq2seq
- blog from wildml #4 Dual Encoder
  - http://www.wildml.com/2016/04/deep-learning-for-chatbots-part-1-introduction/
  - http://www.wildml.com/2016/07/deep-learning-for-chatbots-2-retrieval-based-model-tensorflow/
  - https://github.com/dennybritz/chatbot-retrieval/
- medium https://chatbotsmagazine.com/
- https://github.com/tensorflow/nmt seq2seq translation
  - make use of the new dataset API, can [do batching, padding etc](https://github.com/tensorflow/nmt#data-input-pipeline).
- http://suriyadeepan.github.io/2016-06-28-easy-seq2seq/
- https://blog.kovalevskyi.com/step-by-step-guide-for-creating-a-chatbot-with-sockeye-mxnet-aws-ec2-and-deeplearning-ami-259f443abf3e
- https://github.com/Currie32/Chatbot-from-Movie-Dialogue
  - it is using old non OO version of the seq2seq API  tf1.0

seq2seq

- https://github.com/google/seq2seq
- https://github.com/tensorflow/tensor2tensor
- https://github.com/tensorflow/nmt
- https://github.com/awslabs/sockeye

Dataset

- https://github.com/fendouai/Awesome-Chatbot
- https://breakend.github.io/DialogDatasets/
- https://github.com/Marsan-Ma/chat_corpus
- https://github.com/karthikncode/nlp-datasets
- Ubuntu 2.0 https://github.com/rkadlec/ubuntu-ranking-dataset-creator

Dataset from UCSC

- https://nlds.soe.ucsc.edu/sarcasm2
- https://nlds.soe.ucsc.edu/fc2

Papers

- [Improved Deep Learning Baselines for Ubuntu Corpus Dialogs](https://arxiv.org/abs/1510.03753)
- [Sequential Matching Network: A New Architecture for Multi-turn Response Selection in Retrieval-Based Chatbots](https://github.com/MarkWuNLP/MultiTurnResponseSelection)

Reports

- [stanford cs224n/A Neural Chatbot with Personality](https://web.stanford.edu/class/cs224n/reports/2761115.pdf)

知乎

- [从 Chatbot 到 NER](https://zhuanlan.zhihu.com/p/30845277)
- [完全图解RNN、RNN变体、Seq2Seq、Attention机制](https://zhuanlan.zhihu.com/p/28054589)
- [目前主流的 attention 方法都有哪些](https://www.zhihu.com/question/68482809/answer/268266647)
- [从Encoder到Decoder实现Seq2Seq模型](https://zhuanlan.zhihu.com/p/27608348) 貌似用了新版的 API
  - https://github.com/NELSONZHAO/zhihu/blob/master/basic_seq2seq/Seq2seq_char.ipynb
- [TensorFlow全新的数据读取方式：Dataset API入门教程](https://zhuanlan.zhihu.com/p/30751039)

Existing bots

- **https://github.com/nicolas-ivanov/seq2seq_chatbot_links**
  - https://github.com/Conchylicultor/DeepQA
- [自己动手做聊天机器人教程](https://github.com/warmheartli/ChatBotCourse)
- https://github.com/gunthercox/ChatterBot 4k star, but didn't see much ML code? (seems to be too simple)
- https://github.com/coala/corobo/ a bot for managing open source projects ...
- https://github.com/errbotio/errbot/ plugins for various chat services
- ms bot kit https://docs.microsoft.com/en-us/bot-framework/cognitive-services-bot-intelligence-overview

Distributed training

- https://goodailab.com/how-to-write-distributed-tensorflow-code-with-an-example-on-tensorport-70bf3306adcb
- https://mxnet.incubator.apache.org/how_to/multi_devices.html Model Parallelism vs Data Parallelism
- https://mxnet.incubator.apache.org/how_to/model_parallel_lstm.html
- https://github.com/Microsoft/multiverso/tree/master/Applications/WordEmbedding distributed word embedding

Parameter tunning

- https://github.com/galeone/dynamic-training-bench Simplify the training and tuning of Tensorflow models
- https://blog.rescale.com/deep-neural-network-hyper-parameter-optimization/ (they got a nice UI ....)
- https://github.com/tobegit3hub/advisor (not quite for ML ...) Open-source implementation of Google Vizier for hyper parameters tuning 
- https://docs.travis-ci.com/user/customizing-the-build/#Build-Matrix actually travis has build matrix (which is like grid search ....)

Other

- anime based chatbot https://speakerdeck.com/inureyes/building-ai-chat-bot-using-python-3-and-tensorflow, the source code was supposed to be at https://github.com/inureyes/pycon-apac-2016, but just README ...
